tasks/install.yml

```diff
@@ -1,6 +1,19 @@
 ---
 - name: Install OpenJDK
   apt: name=openjdk-7-jre-headless state=present
+  when: not skip_install and "ansible_os_family" == 'Debian'
+
+- name: Install OpenJDK
```

Comment:
This looks great. One minor change would be to put Debian or Redhat into the names of the tasks. That way when you see one or the other skip during a run it is immediately clear why.